test/FlashtestationRegistry.t.sol

```diff
@@ -395,59 +294,145 @@ contract FlashtestationRegistryTest is Test {
+        bytes memory mockOutput = mockf200.output;
+        bytes memory mockQuote = mockf200.quote;
+        address teeAddress = mockf200.teeAddress;
+
+        // Parse expected report body
+        TD10ReportBody memory expectedReportBody = QuoteParser.parseV4VerifierOutput(mockOutput);
+
+        attestationContract.setQuoteResult(mockQuote, true, mockOutput);
+
+        vm.prank(teeAddress);
+        registry.registerTEEService(mockQuote, mockf200.extData);
+
+        // Get stored report body
+        (bool isValid, IFlashtestationRegistry.RegisteredTEE memory registration) = registry.getRegistration(teeAddress);
+
+        assertTrue(isValid);
+
+        // Verify key fields are stored correctly
+        assertEq(registration.parsedReportBody.mrTd, expectedReportBody.mrTd, "mrTd mismatch");
+        assertEq(registration.parsedReportBody.mrConfigId, expectedReportBody.mrConfigId, "mrConfigId mismatch");
+        assertEq(registration.parsedReportBody.tdAttributes, expectedReportBody.tdAttributes, "tdAttributes mismatch");
+        assertEq(registration.parsedReportBody.xFAM, expectedReportBody.xFAM, "xFAM mismatch");
+    }
+
+    function test_data_extracted_from_reportData() public {
+        // This test verifies that the TEE address is properly extracted from reportData[0:20]
+        bytes memory mockOutput = mockc200.output;
+        bytes memory mockQuote = mockc200.quote;
 
-        // Should not revert and should return a TD10ReportBody
-        TD10ReportBody memory report = registry.getReportBody(teeAddress);
+        // Parse the output to verify the expected address
+        TD10ReportBody memory reportBody = QuoteParser.parseV4VerifierOutput(mockOutput);
+        (address extractedAddress, bytes32 extractedExtDataHash) = QuoteParser.parseReportData(reportBody.reportData);
 
-        assertEq(report.reportData.length, 64, "reportData should be 64 bytes");
-        assertEq(report.reportData, bf42Mock.publicKey, "reportData should match the public key");
+        address expectedAddress = mockc200.teeAddress;
+        assertEq(extractedAddress, expectedAddress, "Address should be extracted from reportData[0:20]");
+
+        require(keccak256(mockc200.extData) == extractedExtDataHash, "Test data mismatch");
+
+        attestationContract.setQuoteResult(mockQuote, true, mockOutput);
+
+        // Can only register from the address in the quote
+        vm.prank(expectedAddress);
+        registry.registerTEEService(mockQuote, mockc200.extData);
+
+        // Verify it was registered to the correct address
+        (bool isValid, IFlashtestationRegistry.RegisteredTEE memory registration) =
+            registry.getRegistration(expectedAddress);
+        assertTrue(isValid);
+        assertEq(registration.extendedRegistrationData, mockc200.extData);
+    }
+
+    function test_extDataMismatch() public {
+        // This test verifies that the TEE address is properly extracted from reportData[0:20]
```

Comment:
this comment is copypasta from the test above, it should say

`// This test verifies that using a mismatched extendedRegistrationData will result in an error